src/main/java/seedu/address/logic/parser/Parser.java

```diff
@@ -13,4 +15,12 @@
      * @throws ParseException if {@code userInput} does not conform the expected format
      */
     T parse(String userInput) throws ParseException;
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    default boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
+        return Stream.of(prefixes).allMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
+    }
```

Comment:
`Parser<T>` is an interface and we normally try to keep our methods abstract. Perhaps you could change this to an abstract class. That or you could still put it in `ParserUtil`, just change the docs header to say it has methods for anything parsing related.

Reply:
If it is too much, you can just revert this to your original implementation within NoteCommandParser, and I'll handle this in my own PR

Reply:
(To be updated later)